moebot_bot/bot/commands/schedule.go

```diff
@@ -0,0 +1,112 @@
+package commands
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/camd67/moebot/moebot_bot/util/db"
+)
+
+type ScheduleCommand struct {
+	schedulers map[db.SchedulerType]Scheduler
+}
+
+func NewScheduleCommand(factory *SchedulerFactory) *ScheduleCommand {
+	return &ScheduleCommand{
+		schedulers: map[db.SchedulerType]Scheduler{
+			db.SchedulerChannelRotation: factory.CreateScheduler(db.SchedulerChannelRotation),
+		},
+	}
+}
+
+func (c *ScheduleCommand) Execute(pack *CommPackage) {
+	if len(pack.params) == 0 {
+		c.listSchedulers(pack)
+		return
+	}
+	if strings.ToUpper(pack.params[0]) == "LIST" {
+		c.listOperations(pack)
+		return
+	}
+
+	if strings.ToUpper(pack.params[0]) == "REMOVE" {
+		c.removeOperation(pack)
+		return
+	}
+	if !c.addOperation(pack) {
+		pack.session.ChannelMessageSend(pack.channel.ID, "Cannot find any scheduler with the command `"+pack.params[1]+"`, please check the commands list.")
+	}
+}
+func (c *ScheduleCommand) GetPermLevel() db.Permission {
+	return db.PermMod
+}
+func (c *ScheduleCommand) GetCommandKeys() []string {
+	return []string{"SCHEDULE"}
+}
+func (c *ScheduleCommand) GetCommandHelp(commPrefix string) string {
+	return fmt.Sprintf("`%[1]s schedule <schedule type> <options>` - Master/Mod Creates a new scheduler with the given type and options. `%[1]s schedule` to list available schedulers.", commPrefix)
+}
+
+func (c *ScheduleCommand) listSchedulers(pack *CommPackage) {
+	message := "List of available schedulers:"
```

Comment:
Switch to [string.builder](https://golang.org/pkg/strings/#Builder) for all string-concat in a loop like this